File: baekjoon/palindrome/1254/Main.java
import java.util.*;

class Main {
    
    static String s;
    static String tmp;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        s = sc.nextLine();
        tmp = s;
        if(isPal(tmp.length()-1)) {
            System.out.println(tmp.length());
        } else {
            for(int i=0; i<s.length(); i++) {
                tmp = s;
                for(int j=i; j>=0; j--) {
                    tmp += s.charAt(j);
                }
                if(isPal(tmp.length()-1)) {
                    System.out.println(tmp.length());
                    return ;
                }
            }
        }
    }
    
    public static boolean isPal(int end) {
        int start = 0;
        while(start < end) {
            if(tmp.charAt(start) != tmp.charAt(end)) {
                return false;    
            }
            start++;
            end--;
        }
        return true;
    }
}

File: baekjoon/palindrome/1254/main.cpp
#include <iostream>
#include <cstdio>
#include <cstring>

using namespace std;

string s;
string tmp;
bool isPal(int end);
int main()
{
    cin >> s;
    tmp = s;
    if(isPal(s.size()-1)) {
        printf("%d\n", (int)tmp.size());
        return 0;
    }
    for(int i=0; i<s.size(); i++) {
        tmp = s;
        for(int j=i; j>=0; j--) {
            tmp.push_back(s[j]);
        }
        if(isPal(s.size()+i)) {
            printf("%d\n", (int)tmp.size());
            return 0;
        }
    }
    
    return 0;
}

bool isPal(int end)
{
    int start = 0;
    while(start < end) {
        if(tmp[start] != tmp[end]) {
            return false;
        }
        start++;
        end--;
    }
    return true;
}